x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/HourOfDay.java

```diff
@@ -30,6 +34,16 @@ protected HourOfDay replaceChild(Expression newChild) {
         return new HourOfDay(source(), newChild, zoneId());
     }
 
+    @Override
+    protected TypeResolution resolveType() {
```

Comment:
Both this method and `doFold` should be implemented on the base class - the subclasses simply repeat the same code over and over.